_posts/2015-07-06-code-reviews-for-learning.md

```diff
@@ -0,0 +1,34 @@
+---
+layout: post
+title: "Why Code Reviews Are Great Learning Tools"
+description: "Code reviews are a good way to make sure (or at least, somewhat more sure) that crapp or buggy code does not make its way to production. But, code reviews are also a terrific way to consolidate your learning. Here's why!"
+tags: ["learning", "resources", "hackership"]
+authors:
+ - anouk
+image: /content/2015-07-06-code-review-whiteboard.jpg
+---
+
+###What is a code review?
+
+A code review is the examination of your code and feedback on your code by a peer, or coach. Code reviews can take place online (for instance by using Github), or in person – offline.
+
+![code reviews are awesome](/content/2015-07-06-code-review-whiteboard.jpg)
+
+###How does a code review work?
+
+Finished a piece of code (or a feature)? Create a patch or pull request and send or assign this to a peer or coach. They will look at your code and make comments, add questions, or give suggestions. 
+
+Alternatively, you could walk someone through your code and explain them what is going on line by line. This is aptly called a code walkthrough.
+
+###Why have a code review?
+
+ 1. **Feedback**
+In general, code reviews are great ways to get feedback on your code and then use this feedback to improve the code. For instance, you may have made a typo or introduced a bug that you are unaware of, maybe a second pair of eyes will help you catch it. Alternatively, a reviewer can give you feedback on style, or give you tips on how to write the same thing in a more efficient way.
+
+ 2. **Knowledge transfer**
+Within teams, code reviews can help you share your knowledge of the code base. In a learning environment code reviews are a good way to update your coach on your progress. Or, alternatively, if your code is reviewed by another learner, they will end up learning something new!
```

Comment:
this also works the other way round: if you review other people's code you can _learn_ more. You find new ways of doing things, learn how other people structure and solve problems and maybe find a language feature you didn't even know about.